backend/src/db/entity-migrations/v3.5.0-add-transformers.ts

```diff
@@ -0,0 +1,79 @@
+import { Knex } from 'knex';
+import { NORTH_ITEMS_TABLE } from '../../repository/north-item.repository';
+import { TRANSFORMERS_TABLE } from '../../repository/transformer.repository';
+import { NORTH_TRANSFORMERS_TABLE } from '../../repository/north-transformer.repository';
+import { SOUTH_TRANSFORMERS_TABLE } from '../../repository/south-transformer.repository';
+import { HISTORY_TRANSFORMERS_TABLE } from '../../repository/history-transformer.repository';
+import { NORTH_CONNECTORS_TABLE } from '../../repository/north-connector.repository';
+import { SOUTH_CONNECTORS_TABLE } from '../../repository/south-connector.repository';
+import { HISTORY_QUERIES_TABLE } from '../../repository/history-query.repository';
+
+export async function up(knex: Knex): Promise<void> {
+  await createNorthItemsTable(knex);
+  await createTransformersTable(knex);
+  await createNorthTransformersTable(knex);
+  await createSouthTransformersTable(knex);
+  await createHistoryTransformersTable(knex);
+}
+
+async function createNorthItemsTable(knex: Knex): Promise<void> {
+  await knex.schema.createTable(NORTH_ITEMS_TABLE, table => {
+    table.uuid('id').primary();
+    table.timestamps(false, true);
+    table.uuid('connector_id').notNullable().references('id').inTable(NORTH_CONNECTORS_TABLE);
+    table.string('name').notNullable();
+    table.boolean('enabled').notNullable();
+    table.json('settings').notNullable();
+
+    table.unique(['connector_id', 'name']);
+  });
+}
+
+async function createTransformersTable(knex: Knex): Promise<void> {
+  await knex.schema.createTable(TRANSFORMERS_TABLE, table => {
+    table.uuid('id').primary();
+    table.timestamps(false, true);
+    table.string('name').notNullable().unique();
+    table.string('description');
+    table.string('input_type').notNullable();
+    table.string('output_type').notNullable();
+    table.string('code').notNullable();
+    table.string('file_regex');
+  });
+}
+
+async function createNorthTransformersTable(knex: Knex): Promise<void> {
+  await knex.schema.createTable(NORTH_TRANSFORMERS_TABLE, table => {
+    table.uuid('north_id').notNullable().references('id').inTable(NORTH_CONNECTORS_TABLE);
+    table.uuid('transformer_id').notNullable().references('id').inTable(TRANSFORMERS_TABLE);
+
+    table.unique(['north_id', 'transformer_id']);
```

Comment:
Not sure this should be unique: we could have transformers associated to a regex in a north connector to parse specific files and another transformer with another regex

Reply:
In that situation we would have two different transformers:
- `transformer_1` with regex
- `transformer_2` without 

and the table would let the values:
`north_1`, `transformer_1`
`north_1`, `transformer_2`

is my understanding correct?